Clarify `CONNECT` permission for getting text in voice messages (#5111)
hammerandchisel/discord-api-docs: docs/resources/Channel.md

## Get Channel Messages % GET /channels/{channel.id#DOCS_RESOURCES_CHANNEL/channel-object}/messages

> Returns the messages for a channel. If operating on a guild channel, this endpoint requires the `VIEW_CHANNEL` permission (and also the `CONNECT` permission for voice channels) to be present on the current user. If the current user is missing the `READ_MESSAGE_HISTORY` permission in the channel then this will return no messages (since they cannot read the message history). Returns an array of [message](#DOCS_RESOURCES_CHANNEL/message-object) objects on success.

Reviewer: ```suggestion
Retrieves the messages in a channel. Returns an array of [message](#DOCS_RESOURCES_CHANNEL/message-object) objects on success.

If operating on a guild channel, this endpoint requires the current user to have the `VIEW_CHANNEL` permission. If the channel is a voice channel, they must _also_ have the `CONNECT` permission.

If the current user is missing the `READ_MESSAGE_HISTORY` permission in the channel, then no messages will be returned.
```

Reviewer: ok i know most of this wording wasn't added by you, but it was getting lengthy and confusing, so i wanted to separate it out a little bit more logically.

Reviewer 2: Yeah that sounds def better and is easier to read

> Returns the messages for a channel. If operating on a guild channel, this endpoint requires the `VIEW_CHANNEL` permission (and also `CONNECT` permission for text in voice channel) to be present on the current user. If the current user is missing the `READ_MESSAGE_HISTORY` permission in the channel then this will return no messages (since they cannot read the message history). Returns an array of [message](#DOCS_RESOURCES_CHANNEL/message-object) objects on success.

Reviewer 3: ```suggestion
Returns the messages for a channel. If operating on a guild channel, this endpoint requires the `VIEW_CHANNEL` permission (and also the `CONNECT` permission for voice channels) to be present on the current user. If the current user is missing the `READ_MESSAGE_HISTORY` permission in the channel then this will return no messages (since they cannot read the message history). Returns an array of [message](#DOCS_RESOURCES_CHANNEL/message-object) objects on success.
```

## Get Channel Message % GET /channels/{channel.id#DOCS_RESOURCES_CHANNEL/channel-object}/messages/{message.id#DOCS_RESOURCES_CHANNEL/message-object}

> Returns a specific message in the channel. If operating on a guild channel, this endpoint requires the `READ_MESSAGE_HISTORY` permission (and also the `CONNECT` permission for voice channels) to be present on the current user. Returns a [message](#DOCS_RESOURCES_CHANNEL/message-object) object on success.

Reviewer: ```suggestion
Retrieves a specific message in the channel. Returns a [message](#DOCS_RESOURCES_CHANNEL/message-object) object on success.

If operating on a guild channel, this endpoint requires the current user to have the `VIEW_CHANNEL` and `READ_MESSAGE_HISTORY` permissions. If the channel is a voice channel, they must _also_ have the `CONNECT` permission.
```

this requires the `VIEW_CHANNEL` permission as well, so adding that here

> Returns a specific message in the channel. If operating on a guild channel, this endpoint requires the `READ_MESSAGE_HISTORY` permission (and also `CONNECT` permission for text in voice channel) to be present on the current user. Returns a [message](#DOCS_RESOURCES_CHANNEL/message-object) object on success.

Reviewer 3: ```suggestion
Returns a specific message in the channel. If operating on a guild channel, this endpoint requires the `READ_MESSAGE_HISTORY` permission (and also the `CONNECT` permission for voice channels) to be present on the current user. Returns a [message](#DOCS_RESOURCES_CHANNEL/message-object) object on success.
```
